pkg/source/source.go

```diff
@@ -241,10 +241,18 @@ func (cs *Channel) syncLoop() {
 	}
 }
 
+// InformerProvider provides an Informer
+type InformerProvider interface {
+	Informer() toolscache.SharedIndexInformer
+}
+
 // Informer is used to provide a source of events originating inside the cluster from Watches (e.g. Pod Create)
 type Informer struct {
-	// Informer is the generated client-go Informer
+	// Informer is the generated client-go Informer.  Mutually exclusive with InformerProvider.
 	Informer toolscache.SharedIndexInformer
+
+	// InformerProvider provides a generated client-go Informer.  Mutually exclusive with Informer.
+	InformerProvider InformerProvider
```

Comment:
`Informer` term is being overused and `source.Informer` and `client-go.Informer` are getting mixed up in my head and causing confusion. Don't have a better suggestions.